examples/meta/src/evaluation/clustering.sg

```diff
@@ -0,0 +1,43 @@
+Math:init_random(1)
+
+File f_feats_train = csv_file("../../../../../data/uci/optdigits/optdigits_feats.dat")
+File f_labels_train = csv_file("../../../../../data/uci/optdigits/optdigits_labels.dat")
+
+#![create_features]
+Features features_train = features(f_feats_train)
+Features labels_train = labels(f_labels_train)
+#![create_features]
+
+#![run_clustering]
+Distance d = distance("EuclideanDistance", lhs=features_train, rhs=features_train)
+Machine kmeans = machine("KMeans", k=10, distance=d)
+kmeans.train()
+RealMatrix centers = kmeans.get_real_matrix("cluster_centers")
+#![run_clustering]
+
+#![assign_labels]
+RealVector labels_vector(10)
+labels_vector[1] = 1
+labels_vector[2] = 2
+labels_vector[3] = 3
+labels_vector[4] = 4
+labels_vector[5] = 5
+labels_vector[6] = 6
+labels_vector[7] = 7
+labels_vector[8] = 8
+labels_vector[9] = 9
+MulticlassLabels labels(labels_vector)
+Features fea_centroids = features(centers)
+Distance d2 = distance("EuclideanDistance", lhs=fea_centroids, rhs=fea_centroids)
+KNN knn(1, d2, labels)
+knn.train()
+MulticlassLabels gnd_hat = knn.apply_multiclass(features_train)
+#![assign_labels]
+
+#![measure_accuracy]
+ClusteringAccuracy accuracy_eval()
```

Comment:
can't call `best_map` if I used base class

Reply:
That’s what we have “wartch_method” for
Check k means base and example

Reply:
@karlnapf do I add the arguments of the method as parameters "watch_parameter"?

Reply:
These cases need refactoring as the `watch_method` functions do not accept arguments.

One option is to require that a caller uses `.put` to set the labels before calling the method (otherwise runtime error with instructions). The other option here would be to use the `.evaluate` method of `CEvaluation` as the signature of the base class is already appropriate here.

Reply:
`ClusteringAccuracy` does not have a method `best_map` btw

Reply:
ah I found it, see discussion in IRC:
```
(14:54:32) Heiko: 1.) the method shouldnt modify the given labels, but rather operate on a copy
(14:54:58) Heiko: 2.) there should be a flag that can disable invocation of the method in ::evaluate
(14:55:07) essam: that's reasonable
(14:55:43) Heiko: 3.) to ensure this is not copy-pasted across all specializations, implement this flag logic in CClusteringEvaluation, and then inside there call a purely virtual method evaluate_impl, which is then overloaded by the sub-classes
```